client/l2tp_client.c

```diff
@@ -282,6 +285,8 @@ l2tp_context *context_new(char *uuid, const char *local_ip, const char *broker_h
   ctx->tunnel_id = tunnel_id;
   ctx->hook = hook ? strdup(hook) : NULL;
 
+  ctx->force_iface = force_iface;
```

Comment:
Can you make a copy of the `force_iface` string - similarly to how it is done for `hook`? And then don't forget to also free `ctx->force_iface` in `context_free`.